cmd/juju/application/bundle/bundle.go

```diff
@@ -345,3 +349,52 @@ func verifyBundle(data *charm.BundleData, bundleDir string, verifyConstraints fu
 	}
 	return errors.Trace(verifyError)
 }
+
+func verifyMixedSeriesBasesMatch(data *charm.BundleData) error {
+	if data.Series != "" && data.DefaultBase != "" {
+		b, err := corebase.ParseBaseFromString(data.DefaultBase)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		s, err := corebase.GetSeriesFromBase(b)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		if s != data.Series {
+			return errors.NotValidf("supplied series %q and base %q in bundle default do not match", data.Series, data.DefaultBase)
+		}
+	}
+
+	for name, m := range data.Machines {
+		if m.Series != "" && m.Base != "" {
+			b, err := corebase.ParseBaseFromString(m.Base)
+			if err != nil {
+				return errors.Trace(err)
+			}
+			s, err := corebase.GetSeriesFromBase(b)
+			if err != nil {
+				return errors.Trace(err)
+			}
+			if s != m.Series {
+				return errors.NotValidf("supplied series %q and base %q in machine %q do not match", m.Series, m.Base, name)
+			}
+		}
+	}
+
+	for name, app := range data.Applications {
+		if app.Series != "" && app.Base != "" {
+			b, err := corebase.ParseBaseFromString(app.Base)
+			if err != nil {
+				return errors.Trace(err)
+			}
+			s, err := corebase.GetSeriesFromBase(b)
+			if err != nil {
+				return errors.Trace(err)
+			}
+			if s != app.Series {
+				return errors.NotValidf("supplied series %q and base %q in application %q do not match", app.Series, app.Base, name)
+			}
```

Comment:
This will add `not valid` at the end, which doesn't make sense. Use `errors.NewNotValid(nil, '...`